lib/byron/test/integration/Test/Integration/Byron/Scenario/API/Transactions.hs

```diff
@@ -440,6 +453,85 @@ scenario_TRANS_CREATE_07 = it title $ \ctx -> do
+        , EncodeAddress n
+        , PaymentAddress n ByronKey
+        )
+    => (Context t -> IO ApiByronWallet)
+    -> SpecWith (Context t)
+scenario_RESTORE_01 fixtureSource = it title $ \ctx -> do
+    -- SETUP
+    let amnt = 100_000 :: Natural
+    wSrc <- fixtureSource ctx
+    (wDest, payment, mnemonics) <- do
+        (wDest, mnemonics) <- fixtureRandomWalletMws ctx
+        let addrs = randomAddresses @n mnemonics
+        pure (wDest, mkPayment @n (head addrs) amnt, mnemonics)
+
+    -- ACTION
+    r <- postByronTransaction @n ctx wSrc [payment] fixturePassphrase
+
+    -- ASSERTIONS
+    let (feeMin, feeMax) = ctx ^. #_feeEstimator $ PaymentDescription
+            { nInputs  = 1
+            , nOutputs = 1
+            , nChanges = 1
+            }
+    verify r
+        [ expectResponseCode HTTP.status202
+        , expectField #amount $ between
+              ( Quantity (feeMin + amnt)
+              , Quantity (feeMax + amnt)
+              )
+        , expectField #direction (`shouldBe` ApiT Outgoing)
+        , expectField #status (`shouldBe` ApiT Pending)
+        ]
+
+    eventually "source balance decreases" $ do
+        rSrc <- request @ApiByronWallet ctx
+            (Link.getWallet @'Byron wSrc) Default Empty
+        verify rSrc
+            [ expectField (#balance . #available) $ between
+                ( Quantity (faucetAmt - amnt - feeMax)
+                , Quantity (faucetAmt - amnt - feeMin)
+                )
+            ]
+
+    eventually "destination balance increases" $ do
+        rDest <- request @ApiByronWallet ctx
+            (Link.getWallet @'Byron wDest) Default Empty
+        verify rDest
+            [ expectField (#balance . #available)
+                (`shouldBe` Quantity (faucetAmt + amnt))
+            ]
+
+    rd1 <- request
+           @ApiByronWallet ctx (Link.deleteWallet @'Byron wDest) Default Empty
+    expectResponseCode @IO HTTP.status204 rd1
+
+    let (Right seed) = fromMnemonic @'[12] (mnemonicToText mnemonics)
+    let rootXPrv = T.decodeUtf8 $ hex $ getKey $
+            generateKeyFromSeed seed
+            (Passphrase $ BA.convert $ T.encodeUtf8 fixturePassphrase)
+    let passwHash =
+            "31347c387c317c2b6a6f747446495a6a566d586f43374c6c54425a576c\
+            \597a425834515177666475467578436b4d485569733d7c78324d646738\
+            \49554a3232507235676531393575445a76583646552b7757395a6a6a2f\
+            \51303054356c654751794279732f7662753367526d726c316c657a7150\
+            \43676d364e6758476d4d2f4b6438343265304b4945773d3d"
+    wDestRestored <- emptyByronWalletFromXPrvWith ctx "random"
+            ("Byron Wallet Restored", rootXPrv, passwHash)
+    eventually "destination balance increases" $ do
+        rDest <- request @ApiByronWallet ctx
+            (Link.getWallet @'Byron wDestRestored) Default Empty
+        verify rDest
+            [ expectField (#balance . #available)
+                (`shouldBe` Quantity (faucetAmt + amnt))
+            ]
```

Comment:
We are missing something crucial here: it doesn't quite matter that one can do a transaction TO an 'XPrv random wallet', but the other way around is much more interesting --> doing a transaction FROM an 'XPrv random wallet' to another wallet. Here, we aren't testing that the key  is actually usable when the passphrase is known (and not usable when it's unknown), so I suggest flipping a bit the steps in this scenario and have:

a) Restore a random fixture wallet from an XPrv (one can get a fixture mnemonic of a funded wallet with `mnemonic <- nextWallet @"random" (_faucet ctx)`

b) Make a transaction from this wallet to another

c) Control that the transaction was successfully executed (i.e., that we were able to decrypt the private key with the given passphrase and use it to sign a transaction!).

Reply:
to be addressed in https://github.com/input-output-hk/cardano-wallet/pull/1476/commits/e3a3d5e08851c5d8c1ead5431fca72c1d80eefed